### PR TITLE
Added HTTP trigger

### LINKED ISSUE #160: Add an HTTP trigger type
**Is your feature request related to a problem? Please describe.**
`wick` needs to be able to serve arbitrary HTTP content from its components.

**Propose a solution**
Add an HTTP trigger that delegates to component operations

### PR DESCRIPTION
**Pull request by jsoverson**
This PR adds an HTTP trigger to wick and contains numerous fixes and changes necessary to support these kinds of triggers.

Closes #160 

feat: adds an HTTP trigger and a raw router that delegates unprocessed http requests to components
feat: improved wick-config JSON schema codegen
feat: added `wick-interface-http`
refactor: moves trigger execution from `wick` to `wick-host`
chore: removed more references to `mesh`
chore: removed `wick-interface-types-keyvalue`
chore: reduced the number of `unused` warnings codegen code can trigger
BREAKING fix: refactored `wick-config` to better reflect wick primitives
BREAKING chore: renamed more `collection` references to `component` and `component` to `operation`
BREAKING fix: updated signature for Component & RpcHandler traits to require +Sync
BREAKING feat: updated `wick-component-codegen` to implement Display and From traits for enums
BREAKING refactor: made v1 and v0 configs in wick-config to prevent using the wrong types
BREAKING refactor: changed wick-host `Host` to `ComponentHost` and added `AppHost`

### CODE REVIEW

File: crates/interfaces/wick-interface-http/interface.json
Comment (jsoverson) [inline crates/interfaces/wick-interface-http/interface.json:1]: This file should be removed

File: crates/wick/wick-host/src/app_host.rs [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+      resources.insert(id.clone(), resource);
+    }
+    Ok(resources)
+  }
+
+  fn start_triggers(&mut self, resources: HashMap<String, Resource>) -> Result<()> {
+    assert!(self.triggers.is_none(), "triggers already started");
+    let resources = Arc::new(resources);
+    let mut triggers = TriggerState::new();
+    for channel_config in self.manifest.triggers() {
```
Comment (jsoverson) [inline crates/wick/wick-host/src/app_host.rs:56]: `channel` refs here should probably be changed to `trigger`

File: crates/wick/wick-runtime/Cargo.toml [hunk 2]
```diff
@@ -38,6 +39,10 @@ tokio-stream = { workspace = true }
 tracing = { workspace = true }
 uuid = { workspace = true }
 atty = { workspace = true }
+hyper = { workspace = true, features = ["runtime", "server"] }
+tower = { workspace = true, features = ["make"], default-features = false }
```
Comment (jsoverson) [inline crates/wick/wick-runtime/Cargo.toml:43]: This dependency isn't used anymore

File: crates/wick/wick-runtime/src/network_service.rs [hunk 3]
```diff
@@ -141,51 +148,49 @@ impl InvocationHandler for NetworkService {
...
-  let namespace = collection.namespace.clone();
-
+  debug!(?collection, ?opts, "initializing component");
+  let namespace = collection.id.clone();
+  let span = opts.span.clone();
   let result = match &collection.kind {
-    ComponentKind::Wasm(v) => initialize_wasm_collection(v, namespace, opts).await,
-    // CollectionKind::GrpcTar(v) => initialize_par_collection(v, namespace, opts).await,
+    ComponentDefinition::Wasm(v) => initialize_wasm_component(v, namespace, opts).instrument(span).await,
     // CollectionKind::GrpcUrl(v) => initialize_grpc_collection(v, namespace).await,
```
Comment (jsoverson) [inline crates/wick/wick-runtime/src/network_service.rs:191]: Uncomment this or remove the ability for a user to configure GRPC components

File: crates/wick/wick-runtime/src/triggers/http/service_factory.rs [hunk 4]
```diff
@@ -0,0 +1,173 @@
...
+                  Ok(b) => {
+                    let _ = tx.send(Packet::encode("body", b));
+                  }
+                  Err(e) => {
+                    let _ = tx.send(Packet::err("body", e.to_string()));
+                  }
+                }
+              }
+              let _ = tx.send(Packet::done("body"));
+              let stream = network.invoke(invocation, rx);
```
Comment (jsoverson) [inline crates/wick/wick-runtime/src/triggers/http/service_factory.rs:145]: This should probably be invoked earlier. As it is, this code waits for the entire body to be read before invoking the operation.

File: crates/wick/wick-runtime/src/network_service.rs (hunk 3 above)
Comment (jsoverson) [reply crates/wick/wick-runtime/src/network_service.rs:191]: Leaving as-is for now. We can address the inconsistency in a followup PR>

**Review by shazaibn — APPROVED** [review]
Looks good!

**Pull request merged**